spec/requests/api/v1/passwords/update_spec.rb

```diff
@@ -0,0 +1,123 @@
+require 'rails_helper'
+
+RSpec.describe 'PATCH /api/v1/users/password', type: :request do
+  let(:user) { create(:user) }
+  let(:password_token) { user.send(:set_reset_password_token) }
+  let!(:headers) do
+    params = {
+      reset_password_token: password_token,
+      redirect_url: ENV['PASSWORD_RESET_URL']
+    }
+    get edit_user_password_path, params: params, headers: auth_headers
+    edit_response_params = Addressable::URI.parse(response.header['Location']).query_values
+    {
+      'access-token' => edit_response_params['token'],
+      'uid' => edit_response_params['uid'],
+      'client' => edit_response_params['client_id']
+    }
+  end
+
+  subject { patch user_password_url, params: params, headers: headers, as: :json }
+
+  context 'when the request is valid' do
+    let(:params) do
+      {
+        "password": 'Newpass123',
+        "password_confirmation": 'Newpass123'
+      }
+    end
+
+    it 'returns the success message' do
+      subject
+      message = I18n.t('devise.passwords.successfully_updated')
+      expect(json).to include_json(message: message)
+    end
+
+    it 'returns a success status' do
+      subject
+      expect(response).to be_successful
+    end
+
+    it 'updates the user reset password token' do
+      expect(user.reload.reset_password_token).to be_present
+      subject
+      expect(user.reload.reset_password_token).to be_nil
+    end
+  end
+
+  context 'when the request is not valid' do
+    context 'when the user does not request reset password' do
+      let(:params) { nil }
+      let(:headers) { nil }
+
+      it 'returns the error messages as a json' do
+        subject
+        expect(json).to include_json(errors: ['Unauthorized'])
+      end
+
+      it 'returns a unauthorized status' do
+        subject
+        expect(response).to have_http_status(:unauthorized)
+      end
+    end
+
+    context 'when params is missing' do
+      let(:params) do
+        {
+          "password_confirmation": 'Newpass123'
```

Comment:
I think you can just put `password_confirmation: 'Newpass123'` (you can avoid putting the double quotes for key values)